daemon/daemon.go

```diff
@@ -482,14 +529,47 @@ func (d *Daemon) Stop() error {
 		d.snapListener.Close()
 	}
 
+	if restartSystem {
+		// give time to polling clients to notice restart
+		time.Sleep(300 * time.Millisecond)
```

Comment:
That feels a bit aggressive. If we're restarting the whole thing, waiting for a few seconds (3?) wouldn't hurt.

Reply:
increased to 3s  (via variable to keep the unit test duration down)